crates/rolldown/src/ecmascript/format/iife.rs

```diff
@@ -201,17 +202,20 @@ fn render_iife_factory_arguments(
     vec![]
   };
   let globals = &ctx.options.globals;
-  externals.iter().for_each(|external| {
-    if let Some(global) = globals.get(external.name.as_str()) {
-      factory_arguments.push(legitimize_identifier_name(global).to_string());
-    } else {
-      let target = legitimize_identifier_name(&external.name).to_string();
-      warnings.push(
-        BuildDiagnostic::missing_global_name(external.name.clone(), ArcStr::from(&target))
-          .with_severity_warning(),
-      );
-      factory_arguments.push(target);
-    }
-  });
+  for external in externals {
+    let global = block_on(globals.call(external.name.as_str()));
```

Comment:
Here using `block_on` is not expected, it could be using `async/await` to do it. But it looks like caused by `SourceJoiner<'source>` is not `Send`. Let us address it at next.

Reply:
Pre-generates these and pass them as args into the `render_iife` function.

Reply:
> Pre-generates these and pass them as args into the `render_iife` function.

I found that doing this would complicate the logic of `render_iife` and `render_umd`, so I simply implemented the `Send` trait for `SourceJoiner`.

Reply:
Good jod, thank you.